### ISSUE TITLE
TeachingBubble header tag need be corrected

### ISSUE THREAD

**Issue by chaofeng2014** [microsoft/fluentui#19136]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)

### Describe the issue:

The header tag in the teaching bubble is using <P>, but actually for accessibility concern and better user experience, we should use `<h1>` or `<h(number)>` tag. So narrator will recognize the tag correctly. 

### Please provide a reproduction of the issue in a codepen:
Screenshot below shows the issue in codepen
![image](https://user-images.githubusercontent.com/6550896/127056252-1f9246de-1818-47d8-a1eb-e89437f13fc7.png)

#### Actual behavior:

The header in the teaching bubble is using `<P>` tag.

#### Expected behavior:

The header in the teaching bubble should use `<H>` tag.

### Documentation describing expected behavior

code here:
https://github.com/jhellesen/office-ui-fabric-react/blob/aede6923ddcce729c56d5cb46338e8569f352687/packages/office-ui-fabric-react/src/components/TeachingBubble/TeachingBubbleContent.base.tsx
line 90:
`const HeaderWrapperAs = typeof headline === 'string' ? 'p' : 'div';`

**Comment by chaofeng2014** [comment]
Maybe because I don't have permission, but this bug should be put label: teachingBubble(this might save some time for bug triage)

**Comment by ling1726** [comment]
@smhigley  could you please take a look at this issue, and see if the heading role is relevant here in the `p` tag ?

**Comment by chaofeng2014** [comment]
Add the accessibility insights for web fast pass result to help understand the accessibility concern:
![image](https://user-images.githubusercontent.com/6550896/127211626-baf51f9c-bbc0-4f38-8e7a-bc57a6aca392.png)

**Comment by smhigley** [comment]
Yes, this should be fixed. Ideally, we would make the `<p>` tag into a real `<h#>` tag, though I'm not sure if that's a good idea for backwards compat. At minimum we should add a prop to customize the heading level, default it to something like 2 or 3, and add `aria-level={level}`.

**Comment by tanisha2710** [comment]
@smhigley  I could take this up, could we discuss a little bit more on this
